### PR TITLE
Sync router state

### LINKED ISSUE #2229: [BUG] View bars swallows first new stage I try to create
On `develop`, the view bar will reliably fail to load a `Limit(10)` stage when I am viewing the `quickstart-geo`, `quickstart-groups`, and `bdd100k` datasets. This does not happen on the `quickstart` dataset. Likely only a problem for larger datasets?

This only happens the first time I try to use the view bar after loading any of these datasets. The next time I try to use the view bar, it works (until I change datasets again).

### PR DESCRIPTION
**Pull request by benjaminpkane**
Resolves #2229. This is a syncing issue with recoil/relay and the url state. @kaixi-wang per you offline comment, Recoil Sync could be a great fit for keeping state in sync. The current implementation isn't ideal.

### CODE REVIEW

**Review by brimoor — CHANGES_REQUESTED** [review]
@benjaminpkane now every time I update the view the entire App reloads. Can you avoid that?

**Comment by brimoor** [comment]
@benjaminpkane I am still seeing the issue pictured above. Also setting `session.view = None` gets me to an invalid state pictured below:

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart-geo")
session = fo.launch_app(dataset)

# Create a Limit(10) stage in the App (whole app refreshes)

session.view = None

# App is now in inconsistent state pictured below
```

Screen Shot 2022-11-04 at 7 46 57 PM

**Comment by kaixi-wang** [comment]
> Resolves #2229. This is a syncing issue with recoil/relay and the url state. @kaixi-wang per you offline comment, Recoil Sync could be a great fit for keeping state in sync. The current implementation isn't ideal.

I'm currently trying to get it working for routing to saved views, but if it works well, we can probably use it for datasets as well.

**Comment by benjaminpkane** [comment]
Things should work now

File: app/packages/state/src/hooks/useSetView.ts [hunk 1]
```diff
@@ -40,6 +41,14 @@ const useSetView = () => {
             },
             onError,
             onCompleted: ({ setView: { dataset, view: value } }) => {
+              console.log(router.history.location.state);
```
Comment (brimoor) [inline app/packages/state/src/hooks/useSetView.ts:44]: remove `console.log`?

**Review by brimoor — APPROVED** [review]
LGTM

**Review by brimoor — CHANGES_REQUESTED** [review]
Hmm okay nevermind, still seeing some state sync issues.

Here's my diagnosis of the GIF:
1. Grid loads wrong number of samples when `limit(5)` view is loaded from Python. I believe the first page of results from the previous state is being shown (not shown in GIF but I cannot scroll past first page). Refreshing in browser resolves the issue
2. Loading `limit(1)` stage now shows first page of previous view's  results in grid
3. Loading `limit(1)` a second time syncs everything correctly, but now the view bar is non-responsive

**Comment by benjaminpkane** [comment]
I can't reproduce any of the above. Can you do a clean install and rebuild. And clear your cache, of course.

`cd ./app && rm -rf node_modules && yarn install && yarn build`

**Comment by brimoor** [comment]
My normal build is to use `install.bash`. I ran `cd ./app && rm -rf node_modules && yarn install && yarn build` this time and am still seeing the issues I reported.

```
sse-starlette==0.10.3
starlette==0.20.4
strawberry-graphql==0.138.1
```

**Comment by benjaminpkane** [comment]
OK, we will have to take this offline. Still unable to reproduce

**Pull request merged**